### PR TITLE
BPMSPL-121 - Tests for UpdateTimerCommand

### PR DESCRIPTION
**Pull request by jakubschwan**
Created tests for UpdateTimerCommand with processes.

### CODE REVIEW

**Comment by triceo** [comment]
Due to the extensive use of timers, the test is going to be extremely unstable. I suggest you find a different way to do this, perhaps using some callback from the process itself.

**Comment by psiroky** [comment]
Yes, we really can not use `Thread.sleep` in the tests. I believe there is a way to use a specific listeners which do get notified when the timer node gets executed. @mswiderski knows more.

**Comment by mswiderski** [comment]
totally agree, we should avoid Thread.sleep completely and try to utilize listeners that are backed by CountDownLatch to be more reliable. An example here: https://github.com/droolsjbpm/jbpm/blob/master/jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/StartEventTest.java#L129

**Comment by mswiderski** [comment]
@jakubschwan do you mind updating the tests to apply mentioned enhancements?

**Comment by jakubschwan** [comment]
Yes, of course. I'm working on it. And thanks for advice

**Comment by jakubschwan** [comment]
Changed tests. There are skipped tests for StartTimer and BoundaryTimer. In this cases UpdateTimerCommand didn't work (didn't change timer), so tests for StartTimer failed and test for BoundaryTimer expired.

File: jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/TimerUpdateTest.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        kieSession.addEventListener(countDownListener);
+        final List<Long> list = new ArrayList<Long>();
+        kieSession.addEventListener(new DefaultProcessEventListener() {
+            @Override
+            public void beforeProcessStarted(ProcessStartedEvent event) {
+                list.add(event.getProcessInstance().getId());
+            }
+        });
+
+        assertEquals(0, list.size());
```
Comment (triceo) [inline jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/TimerUpdateTest.java]: I believe that the project already uses AssertJ in places. Unless the others mind, I'd like this to be ```Assertions.assertThat(list).isEmpty()```. Much clearer in the code, and a very descriptive error message when it fails.

@mswiderski Do you agree?

(Ditto with every other assertion.)

Comment (mswiderski) [inline jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/TimerUpdateTest.java]: agreed

Comment (jakubschwan) [inline jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/TimerUpdateTest.java]: Changed and added to PR

**Comment by mswiderski** [comment]
fixed the boundary event and removed start event tests - see comment in https://github.com/droolsjbpm/jbpm/pull/370 

closing this as https://github.com/droolsjbpm/jbpm/pull/370 includes both commits for tests and fix for boundary timer event.

**Pull request closed without merging**